packages/react/src/components/portal/portal.tsx

```diff
@@ -10,13 +10,20 @@ export interface PortalProps {
 }
 
 export const Portal = (props: PropsWithChildren<PortalProps>) => {
-  const { children, container, disabled } = props
+  const [container, setContainer] = useState(props.container?.current)
+  const { children, disabled } = props
   const isServer = useIsServer()
   const { getRootNode } = useEnvironmentContext()
 
+  useEffect(() => {
+    setContainer(() => {
+      return props.container?.current
+    })
+  }, [props.container?.current])
```

Comment:
The reason the previous implementation passed the tests is because `useIsServer` causes a re-render.

This useEffect emulates the old behavior.